pyproject.toml

```diff
@@ -1,5 +1,166 @@
-# pyproject.toml is currently only used to configure developement tools.
-# Configurations for the lightly package are in setup.py.
+[build-system]
+requires = [
+    "setuptools>=21",
+    "setuptools-scm"
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name="lightly"
+requires-python = ">=3.6"
+authors = [
+    {name = "Lightly Team", email = "team@lightly.ai"},
+]
+license = {file = "LICENSE.txt"}
+description="A deep learning package for self-supervised learning"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Scientific/Engineering :: Image Processing",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "License :: OSI Approved :: MIT License",
+]
+dependencies = [
+  "certifi>=14.05.14",
+  "hydra-core>=1.0.0",
+  "lightly_utils~=0.0.0",
+  "numpy>=1.18.1, <2",
+  "python_dateutil>=2.5.3",
+  "requests>=2.23.0",
+  "six>=1.10",
+  "tqdm>=4.44",
+  "torch",
+  "torchvision",
+  "pydantic >= 1.10.5, < 2",
+  # Note: pytorch_lightning>=1.5 is required for CLI
+  # https://github.com/lightly-ai/lightly/issues/912
+  "pytorch_lightning>=1.0.4",
+  "urllib3 >= 1.25.3",
+  "aenum >= 3.1.11"
+]
+dynamic = ["version", "readme"]
```

Comment:
This is missing description, classifiers, and entry-points.

Reply:
Source: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

Reply:
description and classifiers are specified in:

https://github.com/SauravMaheshkar/lightly/blob/3f542bebdc8ea554ad6c841c4f203042d568a600/pyproject.toml#L15-L35

and the `console_scripts` in the `[project.scripts]` section.